### ISSUE TITLE
Recognise emojis in note titles and show them in the note list

### ISSUE THREAD

**Issue by fran-f** [pbek/QOwnNotes#2190]
Hello there! This is a vague feature request or suggestion, aimed at adding some colour to the Note List, to help scanning and finding notes visually.

At the moment, I can prefix the name of a note with an emoji, for example, "📚 Books". This works great in QOwnNotes, but the emoji becomes part of the filename on disc, creating some trouble in the shell, and for other tools that have difficulties with these characters.

Would it be possible to check if the text of a note opens with an emoji, and show this in the Note List, before the note name? For example, a note names `book-list.md` and opening with this heading
```
📚 Books
======
```
would appear in the Note List as "📚 book-list". This way we could keep filenames simple and compatible with other tools, but help specific notes stand out.

The position of the image *could* be ambiguous, for example if we had another file actually named `📚 stack-list.md`. To avoid this issue, the "promoted" emoji could be rendered as the Note List line icon, replacing the "text placeholder" icon, as in the quick Paint mock below.

![image](https://user-images.githubusercontent.com/50931058/126085515-34364ba6-57fc-4f0b-a4f3-fe913a74e42c.png)

**Comment by pbek** [comment]
Interesting idea, a lot of parts would need to be changed to attempt to make that work (with a lot of possible regressions). And of course  this wouldn't work with Nextcloud Notes...

**Comment by fran-f** [comment]
Agreed, it is a tricky one, and more "delight" that fundamental functionality.

Complexity that I can think of:
- the Note List needs to be capable to render both icons and text (the emoji) in front of each line
- QOwnNotes needs to read the contents, or at least the opening characters, of every note; on sync, on edit, on file change...
- the emoji might need to be stored in a new table in `notes.sqlite`; or maybe it could be implemented as an (invisible?) tag; either way, the Note List would need access to that data

> And of course this wouldn't work with Nextcloud Notes...

Sorry, I am not accustomed with Nextcloud. Do you mean that the emoji would not appear in the same way in Nextcloud, or that this feature would be incompatible with it, breaking sync?

**Comment by pbek** [comment]
> Complexity that I can think of:

plus much more 😁 

> Do you mean that the emoji would not appear in the same way in Nextcloud, or that this feature would be incompatible with it, breaking sync?

I don't think it would break sync. Haven't tested it...

**Comment by fran-f** [comment]
> The hard part, is i.e., figuring out the icon, which I don't think is implementable given the size of our team and complexity of the problem. For example, English is just one language, there are tons of other languages and we don't know how to analyze them.

Oh, I would not expect QOwnNotes to guess the correct icon based on the title of the note. I was just thinking "whatever emoji is already present in the title". So if the note opens with
```
🍰 Cake recipes
================

Sponge cake
-----------
12 eggs
...
```
then the 🍰 emoji would appear in the notes list next to this file.

**Comment by Waqar144** [comment]
That I guess can be done with the help of an item delegate and an emoji font

**Comment by pbek** [comment]
Or "converting" the emoji to an icon (if that's possible) and set it in:

https://github.com/pbek/QOwnNotes/blob/f3bf7ddb3ddd7aa65720bd36908f391c525b6a1b/src/mainwindow.cpp#L2335

**Comment by pbek** [comment]
## 26.7.1

- Added a new checkbox **"Don't use leading emojis in note filename"** in the
  **General settings** (enabled by default): when a note title starts with an
  emoji, the emoji is stripped from the generated filename so filenames stay
  compatible with other tools (for [#2190](https://github.com/pbek/QOwnNotes/issues/2190))
- Added a new checkbox **"Show leading emoji from note title as icon in the note
  list"** in the **Panels settings** (enabled by default): when a note title
  starts with an emoji, the emoji is rendered as the icon for that note in the
  note list and note tree, replacing the standard document icon
  (for [#2190](https://github.com/pbek/QOwnNotes/issues/2190))
